Feature/nfckt 712

alex-kontcur opened the pull request:
1. There should be a function in pharmacy service that creates subscriptions for all KonnektorConfigs the connector

2. There should be a function in the pharmacy service that unsubscribes all KonnektorConfigs

3. The ids of the subscriptions for every konnektor config should be saved in the config folder of this Konnektor

4. The unsubscribe and subscribe function should be callable by the REST services PharmacyResource

alex-kontcur committed:
NFCKT-712

1. There should be a function in pharmacy service that creates subscriptions for all KonnektorConfigs the connector

2. There should be a function in the pharmacy service that unsubscribes all KonnektorConfigs

3. The ids of the subscriptions for every konnektor config should be saved in the config folder of this Konnektor

4. The unsubscribe and subscribe function should be callable by the REST services PharmacyResource

ManuelB commented on src/test/java/health/ere/ps/service/idp/client/IdpClientTest.java:
```diff
@@ -29,8 +14,23 @@
+import java.util.logging.Logger;
 
 @QuarkusTest
+@Disabled("TODO")
```
Why do you disable this test case?

alex-kontcur replied:
Reverted

ManuelB commented on src/test/java/health/ere/ps/service/connector/certificate/CardCertificateReaderServiceTest.java:
```diff
@@ -1,26 +1,26 @@
+import java.util.logging.LogManager;
 
 @QuarkusTest
+@Disabled("TODO")
```
Why do you disable this whole test?

alex-kontcur replied:
Reverted

ManuelB commented on src/test/java/health/ere/ps/service/connector/SingleConnectorServicesProviderTest.java:
```diff
@@ -73,12 +73,14 @@ public void testGetKeyFromBase64OctetDataUri() throws URISyntaxException, KeySto
     }
     
     @Test
+    @Disabled("TODO")
```
This should still work.

ManuelB commented on src/test/java/health/ere/ps/service/connector/SingleConnectorServicesProviderTest.java:
```diff
@@ -1,29 +1,28 @@
 public class SingleConnectorServicesProviderTest {
     private String keystoreFileName = "./src/test/resources/certs/keystore.p12";
-    private String keystoreFilePass = "123456";
+    private String keystoreFilePass = "123456.properties";
```
Why do you change the keystoreFilePass?

alex-kontcur replied:
oops, this is autoreplacement.. reverting

ManuelB commented on src/test/java/health/ere/ps/resource/gematik/ERezeptWorkflowResourceTest.java:
```diff
@@ -75,11 +73,11 @@ public void testIdpToken() {
         when(eRezeptWorkflowResource.httpServletRequest.getHeaderNames()).thenReturn(Collections.enumeration(Collections.emptyList()));
         eRezeptWorkflowResource.eRezeptWorkflowService = mock(ERezeptWorkflowService.class);
-        when(eRezeptWorkflowResource.eRezeptWorkflowService.getBearerToken(any())).thenReturn("123456");
+        when(eRezeptWorkflowResource.eRezeptWorkflowService.getBearerToken(any())).thenReturn("123456.properties");
```
Why do you add .properties to 123456?

alex-kontcur replied:
Reverted

ManuelB commented on src/test/java/health/ere/ps/config/UserConfigTest.java:
```diff
@@ -30,7 +29,7 @@ public void setup() {
         UserConfigurations configurations = userConfig.getConfigurations();
         assertEquals("https://example.com", configurations.getConnectorBaseURL());
-        assertEquals("123456", configurations.getMandantId());
+        assertEquals("123456.properties", configurations.getMandantId());
```
There should not be any .properties

alex-kontcur replied:
Reverted

alex-kontcur committed:
NFCKT-712

5. There should be a scheduled function (https://quarkus.io/guides/scheduler) that renews all 24 hours the subscriptions

5.a The duration should be configurable

6. The amount of subscriptions should be receivable by JMX

7. The subscriptions should be renewed after a Konnektor restart cetp event
